### PR TITLE
qtkeychain 0.8.0 (new formula)

### PR DESCRIPTION
**Pull request by guruz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/qtkeychain.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  def install
+    # We need to set QT_TRANSLATIONS_DIR, else it tries to install translations
+    # into Qt's directory which fails
+    system "cmake", ".", "-DQT_TRANSLATIONS_DIR=#{prefix}/translations", *std_cmake_args
+    system "make", "install"
+    bin.install "testclient"
+  end
+
+  test do
+    assert '#{bin}/testclient 2>&1 | grep account'
```
Comment (bfontaine) [inline Formula/qtkeychain.rb]: Use `assert_match "account", shell_output("#{bin}/testclient 2>&1")` instead.

File: Formula/qtkeychain.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class Qtkeychain < Formula
+  desc "Qt API for storing passwords securely in the keychain"
+  homepage "https://github.com/frankosterfeld/qtkeychain"
+  url "https://github.com/frankosterfeld/qtkeychain/archive/v0.8.0.tar.gz"
+  sha256 "b492f603197538bc04b2714105b1ab2b327a9a98d400d53d9a7cb70edd2db12f"
+
+  depends_on 'cmake' => :build
```
Comment (bfontaine) [inline Formula/qtkeychain.rb]: Use double quotes everywhere.

File: Formula/qtkeychain.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    bin.install "testclient"
+  end
+
+  test do
+    assert '#{bin}/testclient 2>&1 | grep account'
+    # I wanted to do something like this, but this fails because the sandbox
+    # does not have a keychain :-(
+    # system "#{bin}/testclient", "store","__qtkeychaintest", "pw"
+    # assert '#{bin}/testclient restore __qtkeychaintest 2>&1 | grep pw'
+    # system "#{bin}/testclient", "delete","__qtkeychaintest"
```
Comment (bfontaine) [inline Formula/qtkeychain.rb]: You can remove these comments.

File: Formula/qtkeychain.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  depends_on 'cmake' => :build
+  depends_on "qt"
+
+  def install
+    # We need to set QT_TRANSLATIONS_DIR, else it tries to install translations
+    # into Qt's directory which fails
+    system "cmake", ".", "-DQT_TRANSLATIONS_DIR=#{prefix}/translations", *std_cmake_args
+    system "make", "install"
+    bin.install "testclient"
```
Comment (bfontaine) [inline Formula/qtkeychain.rb]: Why is this not installed by `make install`?

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/qtkeychain.rb (hunk 4 above)
Comment (guruz) [inline Formula/qtkeychain.rb]: i guess it's only for development. i only install it to be able to run the brew test. suggestions?

File: Formula/qtkeychain.rb (hunk 2 above)
Comment (guruz) [inline Formula/qtkeychain.rb]: thanks for review, i'll try to fix later today

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@guruz thanks for the PR. Please feel free to open a new one once feedback has been addressed.

**Pull request closed without merging**